File: 26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int k = 0;
        for (int i=1; i< nums.length; i++) {
            if (nums[i] == nums[k]) {
                continue;
            } else {
                k++;
                nums[k] = nums[i];
            }
        }
        return ++k;
    }
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index = 0
        len_nums = len(nums)
        for i in range(1, len_nums):
            if nums[i] != nums[index]:
                index += 1
                nums[index] = nums[i]

        return index + 1


# @lc code=end
